serde-generate/src/java.rs

```diff
@@ -485,28 +488,29 @@ fn output_struct_or_variant_container(
     nested_class: bool,
     package_prefix: &str,
 ) -> Result<()> {
-    let tab = " ".repeat(indentation);
+    let mut out = IndentedWriter::new(out, Indent::Space(4));
+    out.set_level(indentation / 4);
```

Comment:
Now or later, we can simplify the code further by replacing all `out: &dyn Write` occurrences by an indented writer.